### PR TITLE
fix clippy

### PR DESCRIPTION
**Pull request by andrewmwells-amazon**
## Description of changes
Fix clippy errs for not using `cmp` in `partial_cmp` if the type also implements `Ord`. Also fix some warnings.

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)

I confirm that this PR (choose one, and delete the other options):

- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar Dafny model or DRT infrastructure.

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cedar-policy-core/src/ast/value.rs [hunk 1]
```diff
@@ -296,9 +296,7 @@ impl Eq for Set {}
 // HashSet doesn't implement PartialOrd
 impl PartialOrd<Set> for Set {
     fn partial_cmp(&self, other: &Set) -> Option<std::cmp::Ordering> {
-        self.authoritative
-            .as_ref()
-            .partial_cmp(other.authoritative.as_ref())
+        Some(self.cmp(other))
```
Comment (cdisselkoen) [inline cedar-policy-core/src/ast/value.rs:299]: This implementation is here precisely because we want to compare on `authoritative`, not on all of `self` and `other`.  Why this change?

**Review by cdisselkoen — APPROVED** [review]

File: cedar-policy-core/src/ast/value.rs (hunk 1 above)
Comment (andrewmwells-amazon) [reply cedar-policy-core/src/ast/value.rs:299]: Clippy errors without this (See the build from Kesha's PR: https://github.com/cedar-policy/cedar/actions/runs/6421688661/job/17436607880?pr=341)

**Review by khieta — APPROVED** [review]

File: cedar-policy-core/src/ast/value.rs (hunk 1 above)
Comment (shaobo-he-aws) [reply cedar-policy-core/src/ast/value.rs:299]: I think Clippy just tries to ensure the `PartialOrd/Ord` rule: https://doc.rust-lang.org/std/cmp/trait.Ord.html. Calling `cmp` on `self` and `other` doesn't mean a structural comparison.

**Review by shaobo-he-aws — APPROVED** [review]
